DVCLive: Revisit docs organization and content

daavoo opened the pull request:
Closes https://github.com/iterative/dvclive/issues/273
Closes https://github.com/iterative/dvclive/issues/289

Try to focus on the simplest case: Use existing ML integration alongside DVC.

The previous state jumped directly into the API overview and addressed DVC integration on a separate page.
Try to better reflect the high-level steps required to get everything running in the most common case.

shcheklein commented:
I think ppl will start with an existing framework (not an index page of the API). We should be prepared for that as much as we can. Let's say I use Keras and I start from the Keras integration page - how many other pages I'll have to read to get to a working project that I can understand? That's the metric we should be optimizing in the docs here.

Probably, it means that we can even introduce some duplication and / or clear 1-2-3 where 2 and 3 are links to some general pages (dvc.yaml , code sinppet with DVCLive, etc).

daavoo commented:
> I think ppl will start with an existing framework (not an index page of the API). We should be prepared for that as much as we can. Let's say I use Keras and I start from the Keras integration page - how many other pages I'll have to read to get to a working project that I can understand? That's the metric we should be optimizing in the docs here

I was thinking people start from Get Started, but will try to also optimize for this case. 

> Probably, it means that we can even introduce some duplication and / or clear 1-2-3 where 2 and 3 are links to some general pages (dvc.yaml , code sinppet with DVCLive, etc).

I updated Get Started and moved the old one to API Reference. I tried to make the current Get Started a step-by-step with links to more details. Do you think the step-by-step should also be present on each ML Framework page?

shcheklein commented:
so, concept of loggers is more or less clear and familiar I guess, that's why last time I was trying I went right to the Keras page ... I expected to see a simple copy-paste to get started

Get started itself is good for beginners I think in this case.

> Do you think the step-by-step should also be present on each ML Framework page?

I think it would benefit a lot if we make those pages self-containable + easy way to copy paste w/o jumping around and get decent results.

jorgeorpinel commented:
> > I think ppl will start with an existing framework
>
> I was thinking people start from Get Started

Here are top entry pages for DVCLive now:

<img width="567" alt="image" src="https://user-images.githubusercontent.com/1477535/188958112-c2fb3077-0d65-4c07-84da-684dfd5e52d8.png">

Should've probably discussed and planned in https://github.com/iterative/dvclive/issues/273 first (sorry I didn't notice it earlier) but let's check this PR since we have it!

jorgeorpinel commented:
> Added SEO details ☝🏼

jorgeorpinel commented on content/docs/sidebar.json:
```diff
@@ -626,10 +626,6 @@
         "label": "Get Started",
         "slug": "get-started"
       },
-      {
-        "label": "DVCLive with DVC",
-        "slug": "dvclive-with-dvc"
-      },
```
This is a major entry point now. Do we want to burry it under How to/ Integrate with DVC checkpoints? If so let's add a redirect please 🙂 (lmk if you need help with that).

daavoo replied:
Will add a redirect to the current get started. 

I believe the main point of the old page was not about checkpoints but about the need of creating a `dvc.yaml`, which I try to better clarify as a requirement in the current get started

jorgeorpinel commented on content/docs/dvclive/how-to/outputs.md:
```diff
@@ -0,0 +1 @@
+# Track DVCLive outputs
```
What's the plan for this page?

jorgeorpinel replied:
Seems like maybe moving https://dvc.org/doc/dvclive/dvclive-with-dvc#outputs into here is still missing?

jorgeorpinel replied:
Also not sure about term outputs here though.

jorgeorpinel commented on content/docs/dvclive/how-to/checkpoints.md:
```diff
@@ -0,0 +1,42 @@
+# Integrate with DVC Checkpoints
+
+When using [DVC Checkpoints](/doc/user-guide/experiment-management/checkpoints)
+and/or enabling DVCLive's [`resume`](/doc/dvclive/api-reference/live#parameters)
+you need to add the flag
+[`persist: true`](/doc/user-guide/project-structure/pipelines-files#output-subfields)
+to all DVCLive outputs.
```
Probably needs more motivation (why do this/ how is it useful) closer to what we have now in https://dvc.org/doc/dvclive/dvclive-with-dvc#checkpoints.

daavoo replied:
I was trying to make this a concise how-to page. I would expect the first linked page to `user-guide/experiment-management/checkpoints` to cover the motivation for checkpoints and people coming to this page for specific guidance on proper setup, not looking for motivation on the feature

jorgeorpinel commented on content/docs/dvclive/how-to/checkpoints.md:
````diff
@@ -0,0 +1,42 @@
+[`persist: true`](/doc/user-guide/project-structure/pipelines-files#output-subfields)
+to all DVCLive outputs.
+
+Adding `--type checkpoint` to `dvc exp init` will take care of doing this when
+generating the `dvc.yaml`:
+
+```dvc
+$ dvc exp init \
````
Is this the recommended "happy path" method? The intro above is already a bit complicated conceptually (mentioning checkpoints and output persistence), so introducing also experiments isn't ideal. If possible move it after the YAML sample as a tip/ alternative easy way to get there.

daavoo replied:
I changed the order. As commented above the scope here was a concise explanation on how to set up for resume training, not to introduce the concepts to dvclive users

jorgeorpinel replied:
Thanks for addressing my feedback. Sorry I missed the updates until now...

jorgeorpinel commented on content/docs/dvclive/dvclive-with-dvc.md:
```diff
@@ -1,153 +0,0 @@
-# DVCLive with DVC
-
-Even though DVCLive does not require DVC, they can integrate in a couple useful
-ways:
-
-- The [outputs](#outputs) DVCLive produces are recognized by `dvc exp`,
-  `dvc metrics` and `dvc plots`. Those same outputs can be visualized in
-  [Iterative Studio](#iterative-studio).
-
-- DVCLive is also capable of generating [checkpoint](#checkpoints) signal files
-  used by DVC <abbr>experiments<abbr>.
```
This summary was nice. If we remove it, should we mention these things more in DVC docs and link to the new DVCLive how-tos?

daavoo replied:
I believe the first bullet point is now covered in the get started https://dvc-org-dvclive-refacto-czwjzz.herokuapp.com/doc/dvclive/get-started#dvc

Regarding the second bullet point, there is a detailed page about checkpoints in DVC user guide and it already mentions/points to DVCLive.

dberenbaum replied:
Agree with @jorgeorpinel that I liked having a summary of all the DVC magic that DVCLive adds. Otherwise, it feels like readers aren't quite sure what DVCLive is doing.

jorgeorpinel commented on content/docs/dvclive/get-started.md:
```diff
@@ -1,134 +1,129 @@
-## Workflow
-
-### Initialize DVCLive
+In this example we are using [Keras](/doc/dvclive/ml-frameworks/keras) for
+training our model and we can add DVCLive as follows:
```
Note that Hugginfacae and PyTorch seem to be more popular now.

daavoo replied:
I added tabs with keras, huggingface and pytorch lightning . There is a link right bellow to other frameworks

jorgeorpinel commented on content/docs/dvclive/get-started.md:
````diff
@@ -1,134 +1,129 @@
-live.log("acc", 0.9)
-```
+## Set up DVC
 
-See `Live.log()`.
+In order to visualize the outputs generated by DVCLive, we can set up a DVC
+pipeline.
````
Mention this is optional/ recommended?

daavoo replied:
> Mention this is optional/ recommended?

I think we should treat it as mandatory. Given the current capabilities, I don't think it's realistic to expect people using `dvclive` without a `dvc.yaml`.

This might change as we are discussing new "onboarding" workflows, but for now the above is True.

jorgeorpinel replied:
- I wouldn't undersell DVCLive features (working on its own) but up to you. Maybe just mention it can also work without DVC?
- "set up a DVC pipeline" -> I'd avoid the pipeline concept here (and in general). We're talking about a single stage that's just a required wrapper for `dvc exp`.

dberenbaum replied:
I agree with your hunch @daavoo that it's unlikely people want to use DVCLive on its own, but I also agree with @jorgeorpinel that we should briefly mention that DVC is optional. I think it's important for clarity. It currently reads like DVCLive will break if you don't do the DVC setup, and it's unclear why.

daavoo replied:
PTAL to the new version

jorgeorpinel commented on content/docs/dvclive/get-started.md:
```diff
@@ -1,134 +1,129 @@
-<admon type="tip">
+## Set up DVCLive
 
-If you use one of the supported [ML Frameworks](/doc/dvclive/ml-frameworks), you
-can jump directly to its corresponding page.
+First of all, you need to add DVCLive to your Python code in order to start
+logging metrics.
```
I do think this GS approach is better (straight into ML framework integration code sample). But do we still cover generic code application? I guess just in the API ref samples right? (Probably enough)

jorgeorpinel replied:
> do we still cover generic code application? I guess just in the API ref 

I see it's in the [API index page](https://dvc-org-dvclive-refacto-czwjzz.herokuapp.com/doc/dvclive/api-reference) 👍🏼

jorgeorpinel commented on content/docs/dvclive/get-started.md:
```diff
@@ -1,134 +1,129 @@
 
-## Outputs
-
-After you run your training code, you should see the following content in the
-project:
+You can now
+[run experiments](/doc/user-guide/experiment-management/running-experiments)
```
Maybe this should be more generic like "Run your code" and mention the non-DVC workflow as well, even if the examples continue with `dvc exp run`.

jorgeorpinel replied:
Also, do we want to cover plain DVC integration (non-experiments i.e. persistent outputs + `dvc repro`) somewhere?

dberenbaum replied:
I don't think we need to include `dvc repro` since users will get pretty much the same value out of `dvc exp run` here.

jorgeorpinel commented on content/docs/dvclive/get-started.md:
```diff
@@ -1,134 +1,129 @@
-See `Live.log()`, `Live.log_image()` and `Live.log_plot()` for more details.
+After following the above steps we have enabled different ways of monitoring the
+training, comparing, and visualizing the experiments:
 
-### Metrics report
+### DVC
 
-If and when `step` is updated, DVCLive generates or updates a report in
-`dvclive/report.{format}` which will contain all the logged data.
+After the experiment has finished, you can use `dvc exp show` / `dvc plots diff`
+to compare and visualize experiments.
 
-![](/img/dvclive-html.gif)
+While the experiment is running, you can monitor the training with the
+[metrics report](/docs/dvclive/api-reference#metrics-report) that DVCLive
```
Again, I do like this approach a bit more but I feel like it's lacking at least mentioning the vanilla use case (that DVCLive generates a report even if you don't use DVC).

dberenbaum replied:
Yeah, I'd like to see this at the top before you get to DVC. Show that you can run the code normally and get the live metrics report, then move on to DVC.

dberenbaum replied:
It has the added benefit of faster time to value: show something useful (the report) almost immediately.

jorgeorpinel replied:
"faster time to value" is the motto of [Get Started type docs](https://www.notion.so/iterative/Higher-level-content-9408f84a9b694fab949feede39142122#8e6381fcf06643aaa1d8df100fd4014a) 🤩

jorgeorpinel commented on content/docs/dvclive/dvclive-with-dvc.md:
````diff
@@ -1,153 +0,0 @@
-└── train.py
-```
-
-The [metrics summary](/doc/dvclive/api-reference/live/log#description) in
-`training_metrics.json` can be used by `dvc metrics` and visualized with
-`dvc exp show`/`dvc exp diff`.
-
-The [metrics history](/doc/dvclive/api-reference/live/log#step-updates)
-`training_metrics/scalars` can be visualized with `dvc plots`.
-
-The [metrics report](/doc/dvclive/api-reference/live/make_report) in
-`training_metrics/report.html` will contain all the logged data and will be
-automatically updated during training on each `step` update!
````
Are all these documented in the new version?

jorgeorpinel commented on content/docs/dvclive/index.md:
```diff
@@ -8,15 +8,7 @@ fully compatible with DVC. You can
-  <card href="/doc/dvclive/ml-frameworks" heading="ML Frameworks">
-    Use DVCLive alongside your favorite ML Framework
+    A step-by-step introduction
   </card>
```
Not sure I get why we removed these 2 from the docs home page if that's what we want to focus on.

jorgeorpinel commented on content/docs/dvclive/api-reference/index.md:
```diff
@@ -7,5 +7,125 @@ can be imported regularly, for example:
+
+See `Live.next_step()` and `Live.set_step()` for details.
+
+## Putting all together
```
💅🏼 
```suggestion
## Putting it all together
```

jorgeorpinel commented on content/docs/dvclive/api-reference/index.md:
````diff
@@ -7,5 +7,125 @@ can be imported regularly, for example:
+    live.next_step()
+```
+
+## Outputs
+
+After you run your training code, you should see the following content in the
+project:
````
I was never sure about the usage of term "output" here. It can be confusing in the content of DVC (stage `outs`). Maybe "project structure" or "resulting files" or something like that?

daavoo replied:
Renamed to Output folder structure

daavoo replied:
wdyt?

jorgeorpinel replied:
Better, thanks. I still would remove "output", especially since the term is not even used [in the contents](https://dvc.org/doc/dvclive/outputs) (other than the title).

jorgeorpinel commented on content/docs/sidebar.json:
```diff
@@ -676,6 +672,21 @@
+      {
+        "label": "How To",
+        "source": false,
+        "slug": "how-to",
+        "children": [
+          {
+            "slug": "outputs",
+            "label": "Track DVCLive outputs"
+          },
+          {
+            "slug": "checkpoints",
+            "label": "Integrate with DVC checkpoints"
```
Maybe expand the how tos to the top-level since there's only 2. Don't even need to call them "How to". Idea:
**Tracking DVCLive Outputs**
**Using DVC Checkpoints**

dberenbaum commented on content/docs/dvclive/get-started.md:
```diff
@@ -1,134 +1,129 @@
+## Set up DVC
 
-See `Live.log()`.
+In order to visualize the outputs generated by DVCLive, we can set up a DVC
```
Maybe it's better to say that you need DVC in order to track the outputs and compare experiments. You can already visualize the outputs without DVC.

dberenbaum commented on content/docs/dvclive/how-to/checkpoints.md:
```diff
@@ -0,0 +1,42 @@
+# Integrate with DVC Checkpoints
```
Maybe make this more general ("Resume training"?) in case people don't know about DVC checkpoints or are using `resume` without checkpoints.

dberenbaum commented on content/docs/dvclive/how-to/checkpoints.md:
```diff
@@ -0,0 +1,42 @@
+# Integrate with DVC Checkpoints
+
+When using [DVC Checkpoints](/doc/user-guide/experiment-management/checkpoints)
+and/or enabling DVCLive's [`resume`](/doc/dvclive/api-reference/live#parameters)
+you need to add the flag
+[`persist: true`](/doc/user-guide/project-structure/pipelines-files#output-subfields)
+to all DVCLive outputs.
```
```suggestion
When using [DVC Checkpoints](/doc/user-guide/experiment-management/checkpoints)
and/or enabling DVCLive's [`resume`](/doc/dvclive/api-reference/live#parameters)
you need to add the flag
[`persist: true`](/doc/user-guide/project-structure/pipelines-files#output-subfields)
to all DVCLive outputs in `dvc.yaml`.
```

jorgeorpinel commented:
Hi 🙂 

Seems like this may go stale soon. Feel free to break this up into more than one PR so we can get some "less controversial" changes merged @daavoo (if that helps).

Thanks

daavoo commented:
> Hi 🙂
> 
> Seems like this may go stale soon. Feel free to break this up into more than one PR so we can get some "less controversial" changes merged @daavoo (if that helps).
> 
> Thanks

Apologies, I open the P.R. as a side task for the sprint but then I didn't find the time to address the comments.

jorgeorpinel commented:
Not a problem. I just like to check in on inactive PRs now and then. Eventually we'll have to split and merge, or close it for now. No rush though, please prioritize as needed for the product.

dberenbaum commented on content/docs/dvclive/get-started.md:
````diff
@@ -3,164 +3,189 @@
-live.log("acc", 0.9)
-```
+If you prefer to use the DVCLive API directly, check the
+[API Reference](/doc/dvclive/api-reference).
````
Minor: link/mention the basic workflow in the API so it doesn't sound like it's leaving the reader to read through each method.

shcheklein replied:
I think we can add a section to the tabs above "No framework" or "Use Directly" ... ? It's worth doing that. I really love these tabs in the get started!

daavoo replied:
I added a tab named `Python API`, not sure if its a good name

dberenbaum commented on content/docs/dvclive/get-started.md:
````diff
@@ -3,164 +3,189 @@
 
-live.log_param(NUM_EPOCHS)
+```dvc
+$ dvc exp init \
````
```suggestion
$ dvc exp init --explicit \
```

Otherwise I get additional stuff in dvc.yaml compared to what you show below.

dberenbaum commented on content/docs/dvclive/get-started.md:
```diff
@@ -3,164 +3,189 @@
+To build on the previous section, you can optionally wrap our code with DVC to
+run [DVC experiments](/doc/user-guide/experiment-management/) and enable new
+ways to compare and visualize our experiments.
 
```
Hopefully, we can improve this section soon. For now, what do you about making the entire setup an expandable section? Realistically, right now a new user probably needs to go the DVC docs and get set up with DVC before coming back to DVCLive, so I don't think it's worth pushing down the other content to explain how to setup DVC.

daavoo replied:
I still want to highlight the `dvc.yaml` syntax required (`metrics`, `plots`) to get the things rendered. 
Looking at it right now I don't really like that the `dvc.yaml` is hidden in an expandable section.

daavoo replied:
I removed `dvc init` and `exp init` but still displayed the `dvc.yaml`. Still think it makes sense as minimum for building the sections below about different visualization

dberenbaum commented on content/docs/dvclive/get-started.md:
```diff
@@ -3,164 +3,189 @@
+`dvclive/scalars` can be visualized with `dvc plots`.
 
-![](/img/dvclive-html.gif)
+The [metrics summary](/doc/dvclive/api-reference/live/log#description) in
+`dvclive.json` can be visualized with `dvc metrics`.
 
-The `format` can be HTML) or Markdown depending on the value of the `report`
-argument passed to [`Live()`](/doc/dvclive/api-reference/live#parameters).
+If you have logged parameters with `Live.log_param()`, they will be stored in
+`dvclive/params.yaml` and can be visualized with `dvc params`.
 
-<admon type="info">
+In addition, the `dvc exp show` and `dvc exp diff` commands can be used to
+compare both metrics and parameters across experiments.
 
```
I don't think it's worth going through every related DVC command here. Mentioning `dvc exp show` and `dvc plots show/diff` is probably enough.

dberenbaum commented on content/docs/dvclive/get-started.md:
```diff
@@ -3,164 +3,189 @@
+After the experiment has finished and you have committed and pushed the results,
+[Iterative Studio](/doc/studio) will automatically parse the outputs generated
+by DVCLive, allowing you to
+[compare and visualize experiments](/doc/studio/user-guide/projects-and-experiments/visualize-and-compare):
```
To distinguish from the other products above, it might be better to focus on sharing experiments instead of comparing and visualizing.

dberenbaum commented on content/docs/dvclive/resume.md:
```diff
@@ -0,0 +1,40 @@
+# Resume training
+
```
Similar to what @jorgeorpinel said  in https://github.com/iterative/dvc.org/pull/3923#discussion_r965222711, I still think we need more motivation and a high-level summary here, like "DVCLive can enable DVC checkpoints at each step and resume from any previous step."

It would be great if this page could walk through a full example of an experiment with checkpoints and resume.

shcheklein replied:
I would also say that Resume and Outputs look like a lower level sections compared to everything else. 

In DVC they would belong to the User Guide, here I'm not sure.

Especially this one "Resume" feel out-of-the-blue and not clear why we have it, etc

daavoo replied:
Will try to convert it into a more complete guide for resuming training. 
It is just info that is needed in order to properly resume training and didn't fit anywhere else

daavoo replied:
> It would be great if this page could walk through a full example of an experiment with checkpoints and resume.

I ended up dropping the page 👼 . I wrote the guide but it was too much of a duplicate of the existing checkpoints guide https://dvc.org/doc/user-guide/experiment-management/checkpoints#registering-checkpoints-in-your-code. 

For non-checkpoints, I believe we should address proper resuming in DVCLive per https://github.com/iterative/dvclive/issues/140 and document inside each individual framework page.

dberenbaum commented:
Looks really good overall! Mostly need to figure out the resume/checkpoints content and refine the dvc integration a little.

shcheklein commented on content/docs/dvclive/outputs.md:
```diff
@@ -0,0 +1,64 @@
+# Output folder structure
```
Style: here and everywhere - we are using First Letter Capital style for the titles.

shcheklein commented on content/docs/dvclive/get-started.md:
```diff
@@ -3,164 +3,189 @@
 DVCLive is a Python library for logging machine learning parameters, metrics and
 other metadata in simple file formats, which is fully compatible with DVC.
```
Let's mention VS Code ext here?

daavoo replied:
Done

skshetry commented on content/docs/dvclive/api-reference/ml-frameworks/catalyst.md:
````diff
@@ -5,27 +5,19 @@ DVCLive allows you to add experiment tracking capabilities to your
 
-```git
-+from dvclive.catalyst import DvcLiveCallback
+```python
````
@daavoo, you can highlight as before with something like follows:
```suggestion
```python{1,7}
```

You can also highlight a range using someth like `python{1, 4-7}`, etc.

dberenbaum commented on content/docs/dvclive/get-started.md:
```diff
@@ -116,51 +78,98 @@ for epoch in range(NUM_EPOCHS):
+Check the [ML Frameworks](/doc/dvclive/api-reference/ml-frameworks) page for
+more details and other supported frameworks.
+
+If you prefer to use DVCLive directly, check the overview of the
+[Python API](/doc/dvclive/api-reference).
```
Minor/not blocking, but you could drop this now that you added the Python API (which looks good IMO).

dberenbaum commented on content/docs/dvclive/get-started.md:
```diff
@@ -116,51 +78,98 @@ for epoch in range(NUM_EPOCHS):
+run [DVC experiments](/doc/user-guide/experiment-management/) and enable new
+ways to compare and visualize our experiments.
+
+You can find below how an example `dvc.yaml` containing a single sage `train`,
```
```suggestion
You can find below an example `dvc.yaml` containing a single stage `train`,
```

dberenbaum commented on content/docs/dvclive/get-started.md:
```diff
@@ -116,51 +78,98 @@ for epoch in range(NUM_EPOCHS):
+
+![HTML report](/img/dvclive-html.gif).
+
+You can find more details about the structure of files generated by DVCLive in
+the [Output folder structure](/doc/dvclive/outputs).
```
It might be worth showing some example output here, like `dvclive.json` and `dvclive/scalars`, especially since you mention them below without any prior context on this page.

dberenbaum commented on content/docs/dvclive/get-started.md:
````diff
@@ -116,51 +78,98 @@ for epoch in range(NUM_EPOCHS):
+```
 
-### Metrics report
+## Compare and visualize experiments
````
```suggestion
## Compare and visualize DVC experiments
```

Somewhere in the header or text below, I think it's worth clarifying that comparing experiments is part of DVC integration and not standalone DVCLive functionality.

dberenbaum commented on content/docs/dvclive/get-started.md:
```diff
@@ -116,51 +78,98 @@ for epoch in range(NUM_EPOCHS):
+the training, comparing, and visualizing the experiments:
 
-![](/img/dvclive-html.gif)
+### DVC
```
Should this be `CLI` or `DVC CLI`? The VS Code extension is also DVC.

dberenbaum commented on content/docs/dvclive/get-started.md:
```diff
@@ -116,51 +78,98 @@ for epoch in range(NUM_EPOCHS):
 
-</admon>
+![Experiments view](https://github.com/iterative/vscode-dvc/raw/main/extension/resources/walkthrough/images/experiments-table.png)
+![Plost view](https://github.com/iterative/vscode-dvc/raw/main/extension/resources/walkthrough/images/plots-trends.png)
```
```suggestion
![Plots view](https://github.com/iterative/vscode-dvc/raw/main/extension/resources/walkthrough/images/plots-trends.png)
```

dberenbaum commented on content/docs/dvclive/get-started.md:
```diff
@@ -116,51 +78,98 @@ for epoch in range(NUM_EPOCHS):
 
-The `format` can be HTML) or Markdown depending on the value of the `report`
-argument passed to [`Live()`](/doc/dvclive/api-reference/live#parameters).
+You can use `dvc exp show` and `dvc plots` to compare and visualize metrics,
```
Link to https://dvc.org/doc/user-guide/experiment-management/comparing-experiments and/or https://dvc.org/doc/user-guide/experiment-management/visualizing-plots?

dberenbaum reviewed (commented):
Looks great -- only minor comments this time. Once those are addressed, I think it's mergeable.

dberenbaum reviewed (approved):
Thanks @daavoo! Nice improvement here.

jorgeorpinel commented on redirects-list.json:
```diff
@@ -92,6 +92,9 @@
   "^/doc/command-reference/experiments$                                                   /doc/command-reference/exp",
   "^/doc/command-reference/ls$                                                            /doc/command-reference/list 302",
 
+  "^/doc/dvclive/dvclive-with-dvc$                                                        /doc/dvclive/get-started",
+  "^/doc/dvclive/ml-frameworks$                                                           /doc/dvclive/api-reference/ml-frameworks",
```
Update on https://github.com/iterative/dvc.org/pull/3923#issuecomment-1239774786 (relates to the redirects above):

<img width="551" alt="image" src="https://user-images.githubusercontent.com/1477535/197704906-ffbe0b77-85d8-488b-9e78-482b7419fe05.png">

> Looks like frameworks and DVC integration trump the Get Started quite a bit.

Traffic in general is up, esp. to the DVClive's home page and somewhat to the Get Started (👍🏼). But he ML frameworks and DVC integration pages have less traffic now 🤷🏼

jorgeorpinel replied:
p.s. I just realized it's only been a week since this was merged so let's check again later.

jorgeorpinel commented on content/docs/dvclive/get-started.md:
```diff
@@ -1,166 +1,196 @@
-## Workflow
-
-### Initialize DVCLive
+<toggle>
+<tab title="Keras">
```
Is there a criteria to the tabs we include here and their order? Just curious...

<img width="693" alt="image" src="https://user-images.githubusercontent.com/1477535/197716872-63187d5a-528e-49d1-a37f-611f21ff07fb.png">

E.g. rn the top 4 with most traffic are: Pytorch, HugginFace, Tensorflow, Keras (in that order).

jorgeorpinel commented on content/docs/dvclive/get-started.md:
```diff
@@ -1,166 +1,196 @@
-
-<toggle>
-<tab title="Scalars">
+<tab title="Python API">
```
Slightly unclear what this means: 1) All the samples are Python; 2) There's no "Python API" under ML Frameworks. Maybe link to https://dvc.org/doc/dvclive/api-reference/live for this one? Currently they all have the same comment under:

> Check the [ML Frameworks](https://dvc.org/doc/dvclive/api-reference/ml-frameworks) page for more details and other supported frameworks.

jorgeorpinel commented on content/docs/dvclive/get-started.md:
```diff
@@ -1,166 +1,196 @@
+Learn more in the
+[Comparing Experiments](/doc/user-guide/experiment-management/comparing-experiments)
+and [Visualizing Plots](/doc/user-guide/experiment-management/visualizing-plots)
+pages of the user guide.
```
💅🏼 
```suggestion
pages.
```

jorgeorpinel commented on content/docs/dvclive/get-started.md:
```diff
@@ -1,166 +1,196 @@
+![Plots view](https://github.com/iterative/vscode-dvc/raw/main/extension/resources/walkthrough/images/plots-trends.png)
 
-## What next?
+### Share Results
```
Seems like this should be an H2 to me (since it's more about Studio than the DVC integration). This would also create an entry in the right side CONTENTS nav.
